### DISCUSSION TITLE
Question about ParsedMaterial v ParsedFunction

### DISCUSSION THREAD

**Discussion by richmondodufisan** [idaholab/moose#18637] [Q&A General] [answered]
I want a "material property" that is a user defined function of time only. In my input file, under the materials block, I have the following:

```
  [m_inf]
    type = ADParsedMaterial
    f_name = m_inf
    function = 't + 3*t'
    outputs = exodus
  []
```

However I get an error when I try to run this. From the documentation, my understanding is that ParsedMaterial is an "extended" form of ParsedFunction where you can also add variables you're solving for, (e.g displacement) into the function, is that correct? But ParsedFunction only recognizes x,y,z and t. Why then, is this not running?

Also, if I were to use ADParsedFunction, and created this under a "Functions" block, would I still be able to use it in my Material Interface?

Also, when retrieving this from the input file, in my material file, I have:

`_m_inf(getADMaterialPropertyByName<Real>("m_inf")),`

I called it a "Real" since my function is real valued, is that correct? I didn't see anything in the documentation on how to consume ParsedFunction/ParsedMaterials.

**Comment by dschwen** [comment, marked as answer]
> ParsedMaterial is an "extended" form of ParsedFunction 

The two systems are independent and other than the underlying function parser do not share any code. For example there is no `ADParsedFunction`. You cannot use the two interchangably. One is used like any other Function, one is used like any other Material.

**Reply by richmondodufisan** [reply]
Okay, thanks. In my case I want a user-defined function of time only to be used in the Material Interface for computing a material property. Should I be using Parsed Function, then? And if no, what's the reason for the error in the block of code I pasted above? And if yes, how do I consume ParsedFunctions? Since they're not materials, would

`_m_inf(getADMaterialPropertyByName<Real>("m_inf")),` still be correct?

**Reply by GiudGiud** [reply]
You can use a ParsedFunction and a GenericFunctionMaterial 
OR a ParsedMaterial
w/ the appropriate AD pre-fixes to the materials if you want to use AD, but AD does not matter for this material if you only have a time dependence. 

The way you retrieve this material property seems fine (for AD)

**Reply by dschwen** [reply]
or a [`TimeStepMaterial`](https://www.mooseframework.org/source/materials/TimeStepMaterial.html)...
